staging/src/k8s.io/kubectl/pkg/cmd/util/factory.go

```diff
@@ -63,4 +64,7 @@ type Factory interface {
 	Validator(validationDirective string) (validation.Schema, error)
 	// OpenAPISchema returns the parsed openapi schema definition
 	OpenAPISchema() (openapi.Resources, error)
+	// OpenAPIV3Schema returns a client for fetching parsed schemas for
+	// any group version
+	OpenAPIV3Client() (openapiclient.Client, error)
```

Comment:
I think, one day(out of scope of this PR), we need to cut out OpenAPI functions from Factory and instead move them to in new `OpenAPIFactory`. 

Because for example, isn't this new function only used for `kubectl explain`?. By adding this in Factory, we are passing this function into all commands.

Reply:
`OpenAPISchema()` is used throughout, `OpenAPIV3Client()` is used only in explain, but that's just because we haven't wired it further yet. @seans3 is working on replacing some of the `OpenAPISChema()` with `OpenAPIV3Client()` in various places. 

Now that being said, I think the mega factory object is inconvenient to mock in tests, but I don't know how I would refactor that.

Reply:
`Validator`, `OpenAPISchema` and `OpenAPIV3Client` functions can be in `OpenAPIFactory`. All commands might use it something like that;

```go
func NewCmdDiff(f cmdutil.Factory, of cmdutil.OpenAPIFactory, streams genericclioptions.IOStreams) *cobra.Command {
}
```

In my opinion, that will alleviate the responsibilities of mega Factory. If it looks like a good idea, I can propose a PR after @alexzielenski  and @seans3's changes are done.

Reply:
ah, there is already a TODO for it https://github.com/kubernetes/kubernetes/blob/1da46b2f28267421dad9a8e5164e6bce16ae9fcb/staging/src/k8s.io/kubectl/pkg/cmd/util/factory.go#L38 :)